specs/daito/Multi_cloud_yig_integration.md

```diff
@@ -0,0 +1,337 @@
+With scale-out Ceph-cluster and ceph-libradosstripper APIs, YIG solves some of the key performance issues of Ceph
+These features make YIG Ceph Pool an important on-premise Object Storage backend.
+
+### Why YIG:
+YIG brings in some of the features which can complement and enhance OpenSDS multi-cloud management platform. This also brings in opportunity for OpenSDS to be THE Platform for all multi-cloud management. YIG has some features which OpenSDS multi-cloud doesn't have. So YIG can help adding those features into OpenSDS multi-cloud.
+
+ 
+#### How YIG can help in enhancing multi-cloud capabilities of Gelato:
+YIG already supports following features for the YIG Ceph pool:
+
+    * Bucket and Object ACL
+    * Bucket policy
+    * Copying Objects
+    * Server-side encryption
+    * Versioning
+    * Post object
+
+### Goals
+This document will propose the feature enhancement to OpenSDS multi-cloud (Gelato). The new set of features are:
+
+    * Encryption of Data at rest
+    * Bucket Location
+    * Versioning
+    * CORS
+    * ACL
+    * Bucket Policy
+    * Configurable buffered copy object
+    * HEAD Bucket
+    * HEAD Object
+
+### Non-Goals
+N/A
+
+## Design Details
+    
+### Enhance multi-cloud (Gelato) and integrate with YIG
+
+#### For this approach here is the high-level architecture:
+
+![gelato-multicloud-yig](gelato-multicloud-yig.png?raw=true "Gelato Multicloud with YIG")
+
+
+#### Expanding the YIG S3:
+YIG uses TiDB for storing metadata. These metadata are:
+    * Bucket information (includes ACL)
+    * Object Metadata(ACL and contenttype)
+    * Slice information of multipart
+    * Scheduling
+    
+    YIG caches the metadata for improved performance. 'redis' is used for this
+    
+In current form, YIG supports Ceph as the Object storage. 
+OpenSDS multicloud will leverage existing YIG code and extend it to build all the required features.
+Multicloud S3 service flow will remain the same.
+Here is the flow for S3 service:
+    * Client issues an API request
+    * API Gateway receives the request, authenticates and sends it to s3 server
+    * S3 server handles the request. It processes the request on a bucket, updates metadata into DB and cache
+    * S3 server will get backend information from backend service
+    * S3 server will connect with the backend and do the required operations
+    
+
+### Data model impact
+
+Gelato follows the database per service microservices architecture. This means that all the components will have their own database. Gelato microservices architecture currently provides the flexibility, per service, to have their own DB adapter and interact with the respective database for persistent storage.
+With this release, Gelato will support MongoDB for services like Datamover, Dataflow and Backend. 
+S3 service can support both TiDB and MongoDB. YIG project already comes with TiDB support, which can be leveraged.
```

Comment:
For supporting multiple databases, Is it both database support at a time? Are we planning to maintain multiple copies of metadata across these database? If yes, Are we considering data consistency?